_playwright-tests/Integration/AssignTemplateToSystemUI.spec.ts

```diff
@@ -0,0 +1,223 @@
+      await page.getByPlaceholder('Description').fill('Test template for system assignment');
+      await page.getByRole('button', { name: 'Next', exact: true }).click();
+      await page.getByRole('button', { name: 'Create other options' }).click();
+      await page.getByText('Create template only', { exact: true }).click();
+      const rowTemplate = await getRowByNameOrUrl(page, templateName);
+      await expect(rowTemplate.getByText('Valid')).toBeVisible({ timeout: 660000 });
+    });
+
+    await test.step('Boot and register RHSM client', async () => {
+      await regClient.Boot('rhel9');
+      const hostnameResult = await runCmd('Get hostname', ['hostname'], regClient);
+      hostname = hostnameResult?.stdout?.toString().trim() || '';
+      console.log('Container hostname:', hostname);
+
+      const reg = await regClient.RegisterRHC(process.env.ACTIVATION_KEY_1, process.env.ORG_ID_1);
+      if (reg?.exitCode != 0) {
+        console.log('Registration stdout:', reg?.stdout);
+        console.log('Registration stderr:', reg?.stderr);
+      }
+      expect(reg?.exitCode).toBe(0);
+    });
+
+    await test.step('Navigate to template and open Systems tab', async () => {
+      await navigateToTemplates(page);
+      await closePopupsIfExist(page);
+      await expect(page.getByRole('button', { name: templateName })).toBeVisible();
+      await page.getByRole('button', { name: templateName }).click();
+      await expect(page.getByRole('heading', { level: 1 })).toHaveText(templateName);
+      await page.getByRole('tab', { name: 'Systems' }).click();
+      await expect(page.getByRole('button', { name: 'Assign to systems' })).toBeVisible({
+        timeout: 30000,
+      });
+      await page.getByRole('button', { name: 'Assign to systems' }).click();
+      await expect(page.getByText('Assign template to systems')).toBeVisible({ timeout: 30000 });
+      await page.getByRole('searchbox', { name: 'Filter by name' }).fill(hostname);
+      const systemRow = page.getByRole('row').filter({ hasText: hostname });
+      await expect(systemRow).toBeVisible({ timeout: 30000 });
+      await systemRow.getByRole('checkbox').click();
+      await expect(page.getByRole('button', { name: 'Assign', exact: true })).toBeVisible({
+        timeout: 30000,
+      });
+      await page.getByRole('button', { name: 'Assign', exact: true }).click();
+      await expect(page.getByText('Template successfully added to 1 system')).toBeVisible({
+        timeout: 30000,
+      });
+      await expect(page.getByRole('tab', { name: 'Systems' })).toHaveAttribute(
+        'aria-selected',
+        'true',
+      );
+      await expect(
+        page.getByRole('grid', { name: 'assign systems table' }).getByText(hostname),
+      ).toBeVisible({ timeout: 30000 });
+      await expect(systemRow).toBeVisible({ timeout: 30000 });
+    });
+
+    await test.step('Verify the host can install packages from the template', async () => {
```

Comment:
No rhc refresh and anything? If this works I would be slightly suspicious - maybe it can install that package even when not in template? If it works as designed cool!